Client/src/Components/SessionTimeout.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext, useState } from "react";
+
+import { Alert } from '@material-ui/lab';
+import { Snackbar } from '@material-ui/core';
+import { UserContext } from "../App";
+import { defaultUserAccountModel } from '../Models/Authentication/IUserAccountModel';
+
+import useInterval from 'react-useinterval';
+import { endUserSession } from "../Common/GenericHelpers";
+import { useHistory } from "react-router";
+import { loginRoute } from "../Common/Consts/Routes";
+
+export const SessionTimeOut = () => {
+
+    const { user, setUser } = useContext(UserContext);
+    const [seconds, setSeconds] = React.useState(user.sessionExpiration);
+    const [open, setOpen] = useState<boolean>(false)
+    const history = useHistory()
+
+    useInterval(() => {
+        if (seconds === 1) {
+            redirectToLogin()
+        }
+        else if (seconds < 30) {
```

Comment:
i dont think 30 seconds is enough for the user to react to being logged out, save their progress and relogin, it should be in range of minutes, like 5 minutes maybe

Reply:
I agree, there is no sensitive information so the session can be 5 minutes

Reply:
Done